packages/citizen-sdk/README-ClaimSDK.md

```diff
@@ -0,0 +1,228 @@
+import { useIdentitySDK } from '@goodsdks/identity-sdk/wagmi-sdk';
+import { ClaimSDK } from '@goodsdks/identity-sdk/viem-claim-sdk';
+
+const ClaimComponent = () => {
+  const { address } = useAccount();
+  const publicClient = usePublicClient();
+  const { data: walletClient } = useWalletClient();
+  const identitySDK = useIdentitySDK('production');
+
+  if (!address || !publicClient || !walletClient || !identitySDK) {
+    return <div>Loading...</div>;
+  }
+
+  const claimSDK = new ClaimSDK({
+    account: address,
+    publicClient,
+    walletClient,
+    identitySDK,
+    env: 'production',
+  });
+
+  // Use claimSDK to call methods like claim(), checkEntitlement(), etc.
+  return <div>Claim SDK Initialized</div>;
+};
+```
+
+### Using the Viem SDK for Claim
+
+For non-React environments or backend services, you can initialize the Claim SDK directly using Viem's clients. The `ClaimSDK.init` method simplifies the process by fetching the account address from the wallet client.
+
+```typescript
+import { PublicClient, WalletClient } from "viem"
+import { IdentitySDK } from "@goodsdks/identity-sdk/viem-identity-sdk"
+import { ClaimSDK } from "@goodsdks/identity-sdk/viem-claim-sdk"
+
+const publicClient = new PublicClient({
+  /* configuration */
+})
+const walletClient = new WalletClient({
+  /* configuration */
+})
+const identitySDK = new IdentitySDK(publicClient, walletClient, "production")
+
+const claimSDK = await ClaimSDK.init({
+  publicClient,
+  walletClient,
+  identitySDK,
+  env: "production",
+})
+```
+
+## API Reference
+
+### Viem SDK
+
+#### `ClaimSDK` Class
+
+The `ClaimSDK` class manages interactions with the UBI Scheme Contract to enable users to claim UBI. It requires an instance of `IdentitySDK` for whitelisting checks.
+
+**Constructor:**
+
+```typescript
+new ClaimSDK({
+  account: Address,
+  publicClient: PublicClient,
+  walletClient: WalletClient<any, Chain | undefined, Account | undefined>,
+  identitySDK: IdentitySDK,
+  rdu?: string,
+  env?: contractEnv
+})
+```
+
+**Parameters:**
+
+- `account`: The user's wallet address.
+- `publicClient`: A Viem `PublicClient` instance for reading blockchain data.
+- `walletClient`: A Viem `WalletClient` instance for sending transactions.
+- `identitySDK`: An instance of `IdentitySDK` for identity verification.
+- `rdu` _(optional)_: Redirect URL for face verification, defaults to the current window location.
+- `env` _(optional)_: Environment setting (`"production" | "staging" | "development"`), defaults to `"production"`.
+
+**Static Method:**
+
+- **`init(props: Omit<ClaimSDKOptions, "account">): Promise<ClaimSDK>`**
+
+  Asynchronously initializes the Claim SDK by retrieving the account from the wallet client.
+
+  ```typescript
+  const claimSDK = await ClaimSDK.init({
+    publicClient,
+    walletClient,
+    identitySDK,
+    env: "production",
+  })
+  ```
+
+**Methods:**
+
+- **`checkEntitlement(pClient?: PublicClient): Promise<bigint>`**
```

Comment:
why does it accept pclient as input?

Reply:
see comment here: https://github.com/GoodDollar/GoodSDKs/pull/3#discussion_r2095344617